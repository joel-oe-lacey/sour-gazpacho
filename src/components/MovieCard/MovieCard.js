import React from 'react';
import { Link } from 'react-router-dom';
import './MovieCard.scss';


const MovieCard = ({ id, title, userRating, averageRating, poster }) => {
    return (
        <article className="movie-card">
          <Link to={`/movies/${id}`}><img src={poster} alt='movie poster' className='movie-poster'/></Link>
          <div className='movie-info'>
            <h3 className='movie-title'>{title}</h3>
            <h4 className='movie-rating'>Average Rating: {(averageRating).toFixed(1)}</h4>
            {userRating && <h4 className='movie-rating'>Your Rating: {averageRating}</h4>}
          </div>
        </article>
    )
}

export default MovieCard;
