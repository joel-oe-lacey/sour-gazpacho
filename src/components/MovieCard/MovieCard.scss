.movie-card {
  width: 310px;
  height: 410px;
  border: solid 5px #8E0204;
  margin: 10px;
  color: rgba(0,0,0,0);
}

.movie-poster {
  height: 400px;
  width: 300px;
  object-fit: cover;
  z-index: 1;
}

.movie-card:hover {
  color: #FFF;
}

.movie-title {
  text-align: center;
  z-index: 10;
  background-color: #8E0204;
  width: 300px;
  margin-top: -100px;
}

.movie-rating {
  text-align: center;
  z-index: 10;
  background-color: #8E0204;
}
