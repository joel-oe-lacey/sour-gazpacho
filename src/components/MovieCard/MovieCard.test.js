import React from 'react';
import MovieCard from './MovieCard.js';
import { shallow } from 'enzyme';

describe('MovieCard', () => {
let wrapper;

  beforeEach(() => {
    wrapper = shallow(<MovieCard
      id={22}
      title='Parasite'
      poster='https:/image.some.jpg'
      averageRating={5.0}
      userRating={null}
      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

})
