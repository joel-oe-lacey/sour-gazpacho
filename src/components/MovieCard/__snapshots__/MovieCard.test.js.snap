// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard should match the snapshot 1`] = `
<article
  className="movie-card"
>
  <Link
    to="/movies/22"
  >
    <img
      alt="movie poster"
      className="movie-poster"
      src="https:/image.some.jpg"
    />
  </Link>
  <div
    className="movie-info"
  >
    <h3
      className="movie-title"
    >
      Parasite
    </h3>
    <h4
      className="movie-rating"
    >
      AverageRating: 
      5.0
    </h4>
  </div>
</article>
`;
